tests/snuba/api/endpoints/test_organization_events.py

```diff
@@ -5723,6 +5729,27 @@ def test_floored_epm_more_events(self):
         assert data[0]["epm()"] == 12.5
         assert data[0]["floored_epm()"] == 10
 
+    def test_user_misery_using_discover(self) -> None:
+        """This test is used to determine data differences between on-demand and this."""
+        self._setup_user_misery()
+
+        query = {"field": ["user_misery(300)"]}
+        response = self.do_request(query)
+
+        assert response.status_code == 200, response.content
+        assert response.data == {
+            "data": [{"user_misery(300)": user_misery_formula(miserable_users=3, unique_users=4)}],
```

Comment:
In the on-demand test, we will be iterating over the same data structure, yet, it will not pass.